### PR TITLE
[SPARK-43230][CONNECT] Simplify `DataFrameNaFunctions.fillna`

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
add a helper function in `DataFrameNaFunctions`

### Why are the changes needed?
to Simplify `DataFrameNaFunctions.fillna`

### Does this PR introduce _any_ user-facing change?
no, dev-only

### How was this patch tested?
existing UTs

### CODE REVIEW

**Review by amaliujia — APPROVED** [review]
LGTM

**Comment by zhengruifeng** [comment]
```
ERROR: Comparing client jar: /__w/spark/spark/connector/connect/client/jvm/target/scala-2.12/spark-connect-client-jvm-assembly-3.5.0-SNAPSHOT.jar and and sql jar: /__w/spark/spark/sql/core/target/scala-2.12/spark-sql_2.12-3.5.0-SNAPSHOT.jar 
problems: 
method fillValue(java.lang.Object,scala.Option)org.apache.spark.sql.Dataset in class org.apache.spark.sql.DataFrameNaFunctions does not have a correspondent in client version
Exceptions to binary compatibility can be added in 'CheckConnectJvmClientCompatibility#checkMiMaCompatibility'
connect-client-jvm module mima check failed.
Error: Process completed with exit code 1.
```

seems Mima also check `private[sql]`

File: connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/connect/client/CheckConnectJvmClientCompatibility.scala [hunk 1]
```diff
@@ -145,6 +145,7 @@ object CheckConnectJvmClientCompatibility {
 
       // DataFrameNaFunctions
       ProblemFilters.exclude[Problem]("org.apache.spark.sql.DataFrameNaFunctions.this"),
+      ProblemFilters.exclude[Problem]("org.apache.spark.sql.DataFrameNaFunctions.fillValue"),
```
Comment (zhengruifeng) [inline connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/connect/client/CheckConnectJvmClientCompatibility.scala:148]: @LuciferYang  it seems the MiMa tests also check `private[sql]`, is this as expected?

Comment (LuciferYang) [reply connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/connect/client/CheckConnectJvmClientCompatibility.scala:148]: This behavior is expected now, but I think some refactoring work is needed to automatically filter similar checks, just like what `dev/mima` check does.

Comment (zhengruifeng) [reply connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/connect/client/CheckConnectJvmClientCompatibility.scala:148]: got it, thanks

**Review by LuciferYang — APPROVED** [review]

File: connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/connect/client/CheckConnectJvmClientCompatibility.scala (hunk 1 above)
Comment (LuciferYang) [reply connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/connect/client/CheckConnectJvmClientCompatibility.scala:148]: filed https://issues.apache.org/jira/browse/SPARK-43246 to tracking this

**Comment by zhengruifeng** [comment]
Thanks for the reviews, merged to master

**Pull request closed without merging**